### PR TITLE
Refactored several samples

### PR DESCRIPTION
**Pull request by mshabunin**
### This pullrequest changes

- simplified and refactored some samples
- added license headers to these files

### CODE REVIEW

File: samples/tapi/dense_optical_flow.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        parser.printErrors();
+        return 1;
+    }
+
+    VideoCapture cap;
+    if(camera != -1)
+        cap.open(camera);
+    else
+        cap.open(filename, CAP_FFMPEG);
```
Comment (alalek) [inline samples/tapi/dense_optical_flow.cpp]: > CAP_FFMPEG

It is not necessary.

File: samples/dnn/colorization.cpp [hunk 2]
```diff
@@ -43,53 +41,60 @@ float hull_pts[] = {
...
-        "{ opencl         | false | enable OpenCL }"
-    );
-
-    String modelTxt = parser.get<string>("proto");
-    String modelBin = parser.get<string>("model");
-    String imageFile = parser.get<String>("image");
-    if (parser.get<bool>("help") || modelTxt.empty() || modelBin.empty() || imageFile.empty())
+    const string about =
+        "This sample demonstrates recoloring grayscale images with dnn.\n"
+        "This program is based on:\n\n"
```
Comment (alalek) [inline samples/dnn/colorization.cpp]: `\n\n` -> `\n` ?

File: samples/cpp/peopledetect.cpp [hunk 3]
```diff
@@ -1,177 +1,131 @@
...
+        // (and more false alarms, respectively), decrease the hitThreshold and
+        // groupThreshold (set groupThreshold to 0 to turn off the grouping completely).
+        vector<Rect> found;
+        if (m == Default)
+            hog.detectMultiScale(img, found, 0, Size(8,8), Size(32,32), 1.05, 2, false);
+        else if (m == Daimler)
+            hog_d.detectMultiScale(img, found, 0.5, Size(8,8), Size(32,32), 1.05, 2, true);
+        return found;
+    }
+    static void adjustRect(Rect & r)
```
Comment (alalek) [inline samples/cpp/peopledetect.cpp]: BTW, "static" method is used via object: `detector.adjustRect(r);`

File: samples/dnn/colorization.cpp [hunk 4]
```diff
@@ -43,53 +41,60 @@ float hull_pts[] = {
...
+    parser.about(about);
+    if (parser.has("help"))
     {
-        cout << "A sample app to demonstrate recoloring grayscale images with dnn." << endl;
         parser.printMessage();
         return 0;
     }
+    string modelTxt = parser.get<string>("proto");
+    string modelBin = parser.get<string>("model");
+    string imageFile = parser.get<String>("image");
```
Comment (alalek) [inline samples/dnn/colorization.cpp]: `String/string` mess.

**Comment by mshabunin** [comment]
@alalek , maybe it will be better to add special function to the _highgui_ module which will wait for user input by default and can be controlled by some environment variable and command line switches to allow non-interactive execution:
```.cpp
bool isInteractive = parser.get<bool>("interactive");
// or set environment variable OPENCV_INTERACTIVE=0 to force batch execution
...
cv::waitForUser("Press any key to exit...", !isInteractive);
```

**Pull request merged**